packages/adapter/src/lib/adapter/utils.ts

```diff
@@ -125,3 +126,20 @@ export async function listInstalledNodeModules(namespace: string): Promise<strin
 
     return dependencies;
 }
+
+/**
+ * Request a module name by given url using `npm view`
+ *
+ * @param url the url to the package which should be installed via npm
+ */
+export async function requestModuleNameByUrl(url: string): Promise<string> {
+    const res = await execAsync(`npm view ${url} name`);
```

Comment:
any need to set a cwd?

Reply:
Nope, not using any locally installed stuff